validation/src/main/scala/hmda/validation/rules/ts/syntactical/S025.scala

```diff
@@ -20,10 +20,6 @@ class S025 private (institution: Institution) extends EditCheck[HasControlNumber
   def apply(input: HasControlNumber): Result = compare(input.respondentId, input.agencyCode)
 
   private def compare(filingRespId: String, filingAgencyCode: Int): Result = {
-    institution.respondentId match {
-      case Left(invalid) => Failure()
-      case Right(validRespId) =>
-        (filingRespId is equalTo(validRespId.id)) and (filingAgencyCode is equalTo(institution.agency.value))
-    }
+    (filingRespId is equalTo(institution.respondent.id.id)) and (filingAgencyCode is equalTo(institution.agency.value))
```

Comment:
This reads a little weird and looks like a typo, though it isn't. Maybe rename respondent id to externalId?